src/lib.rs

```diff
@@ -274,9 +275,92 @@ fn parse_unary_operator(input: ParseInput) -> ParseResult<UnaryOperator> {
     spanned(op).parse(input)
 }
 
-/// Parses a unary expression. E.g., `-u1:1`, `!u1:1`
-fn parse_unary_expression(input: ParseInput) -> ParseResult<(UnaryOperator, Expression)> {
-    tuple((parse_unary_operator, parse_expression)).parse(input)
+/// Parses unary and atomic expressions. E.g., `-u1:1`, `u1:1`
+fn parse_unary_atomic_expression(input: ParseInput) -> ParseResult<Expression> {
+    // this implementation follows the 'Top Down Operator Precedence' algorithm. See
+    // <https://btmc.substack.com/p/how-to-parse-expressions-easy> or <https://tdop.github.io/>
+    alt((
+        spanned(parse_literal),
+        // TODO parse parenthesized expressions? see
+        // https://btmc.substack.com/p/how-to-parse-expressions-easy
+        spanned(tuple((parse_unary_operator, parse_unary_atomic_expression))),
+    ))
+    .parse(input)
+}
+
+/// Parses a binary operator and the expression that follows it, given the expression preceding
+/// the operator, returning all of this is a binary `Expression`.
+///
+/// E.g. left=`u1:1`, input=`&& u1:1`, returns the Expression for `u1:1 && u1:1`
+///
+/// parse_infix_expression handles any and all kinds of expressions that would
+/// left-recursively call parse_expression
+fn parse_infix_expression(input: ParseInput, left: Expression) -> ParseResult<Expression> {
+    // this implementation follows the 'Top Down Operator Precedence' algorithm. See
```

Comment:
These 3 comment lines probably aren't necessary, given that they're in the parse_expression code below.

Reply:
FWIW I prefer that this comment appears in all the 3 functions that are core to the TDOP algorithm.